Building macOS dmg with Zulu+FX JVM fails

### Please agree to the following

- [X] I have searched [existing issues](https://github.com/cryptomator/cryptomator/issues?q=) for duplicates
- [X] I agree to follow this project's [Code of Conduct](https://github.com/cryptomator/cryptomator/blob/develop/.github/CODE_OF_CONDUCT.md)

### Summary

Current macOS CI build fail, see https://github.com/cryptomator/cryptomator/actions/runs/5646443055/job/15317844949#step:8:56.

### What software is involved?

- Operating System: macOS Github runner
- Cryptomator: 1.10.0-beta1


### Volume Type

None

### Steps to Reproduce

1. Run macOS build


### Expected Behavior

Builds produces dmg

### Actual Behavior

Build fails while building custom JVM/ executing jlink

### Reproducibility

Always

### Relevant Log Output

```shell
Run ${JAVA_HOME}/bin/jlink --verbose --output runtime --module-path "${JAVA_HOME}/jmods" --add-modules java.base,java.desktop,java.instrument,java.logging,java.naming,java.net.http,java.scripting,java.sql,java.xml,javafx.base,javafx.graphics,javafx.controls,javafx.fxml,jdk.unsupported,jdk.crypto.ec,jdk.accessibility,jdk.management.jfr --strip-native-commands --no-header-files --no-man-pages --strip-debug --compress=1

  ${JAVA_HOME}/bin/jlink --verbose --output runtime --module-path "${JAVA_HOME}/jmods" --add-modules java.base,java.desktop,java.instrument,java.logging,java.naming,java.net.http,java.scripting,java.sql,java.xml,javafx.base,javafx.graphics,javafx.controls,javafx.fxml,jdk.unsupported,jdk.crypto.ec,jdk.accessibility,jdk.management.jfr --strip-native-commands --no-header-files --no-man-pages --strip-debug --compress=1
  shell: /bin/bash -e {0}
  env:
    JAVA_VERSION: 20
    JAVA_HOME: /Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64
    JAVA_HOME_20_X64: /Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64
java.base file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.base.jmod
java.datatransfer file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.datatransfer.jmod
java.desktop file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.desktop.jmod
java.instrument file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.instrument.jmod
java.logging file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.logging.jmod
java.management file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.management.jmod
java.naming file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.naming.jmod
java.net.http file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.net.http.jmod
java.prefs file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.prefs.jmod
java.scripting file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.scripting.jmod
java.security.sasl file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.security.sasl.jmod
java.sql file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.sql.jmod
java.transaction.xa file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.transaction.xa.jmod
java.xml file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/java.xml.jmod
javafx.base file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/javafx.base.jmod
javafx.controls file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/javafx.controls.jmod
javafx.fxml file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/javafx.fxml.jmod
javafx.graphics file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/javafx.graphics.jmod
jdk.accessibility file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/jdk.accessibility.jmod
jdk.crypto.ec file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/jdk.crypto.ec.jmod
jdk.jfr file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/jdk.jfr.jmod
jdk.management file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/jdk.management.jmod
jdk.management.jfr file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/jdk.management.jfr.jmod
jdk.unsupported file:///Users/runner/hostedtoolcache/Java_Zulu_jdk+fx/20.0.2-9/x64/jmods/jdk.unsupported.jmod

Providers:
  java.desktop provides java.net.ContentHandlerFactory used by java.base
  java.base provides java.nio.file.spi.FileSystemProvider used by java.base
  java.naming provides java.security.Provider used by java.base
  java.security.sasl provides java.security.Provider used by java.base
  jdk.crypto.ec provides java.security.Provider used by java.base
  java.base provides java.util.random.RandomGenerator used by java.base
  java.desktop provides javax.print.PrintServiceLookup used by java.desktop
  java.desktop provides javax.print.StreamPrintServiceFactory used by java.desktop
  java.management provides javax.security.auth.spi.LoginModule used by java.base
  java.desktop provides javax.sound.midi.spi.MidiDeviceProvider used by java.desktop
  java.desktop provides javax.sound.midi.spi.MidiFileReader used by java.desktop
  java.desktop provides javax.sound.midi.spi.MidiFileWriter used by java.desktop
  java.desktop provides javax.sound.midi.spi.SoundbankReader used by java.desktop
  java.desktop provides javax.sound.sampled.spi.AudioFileReader used by java.desktop
  java.desktop provides javax.sound.sampled.spi.AudioFileWriter used by java.desktop
  java.desktop provides javax.sound.sampled.spi.FormatConversionProvider used by java.desktop
  java.desktop provides javax.sound.sampled.spi.MixerProvider used by java.desktop
  java.logging provides jdk.internal.logger.DefaultLoggerFinder used by java.base
  java.desktop provides sun.datatransfer.DesktopDatatransferService used by java.datatransfer
  jdk.management provides sun.management.spi.PlatformMBeanProvider used by java.management
  jdk.management.jfr provides sun.management.spi.PlatformMBeanProvider used by java.management
Error: jdk.tools.jlink.plugin.PluginException: Duplicate resources: {conf/management/management.properties=[jdk.management.jfr, jdk.management, jdk.unsupported], lib/javafx.properties=[javafx.base, javafx.controls, javafx.fxml], lib/libinstrument.dylib=[java.logging, java.instrument], conf/logging.properties=[java.sql, java.logging, java.transaction.xa, java.management, java.xml, jdk.jfr, javafx.base, java.naming, javafx.controls, javafx.graphics, java.scripting, javafx.fxml, java.prefs, java.net.http, jdk.accessibility, java.security.sasl, jdk.crypto.ec], lib/libj2pcsc.dylib=[java.sql, java.transaction.xa, java.xml], conf/sound.properties=[java.desktop, java.instrument], conf/management/jmxremote.access=[jdk.management.jfr, jdk.management, jdk.unsupported], lib/libmanagement.dylib=[java.naming, java.management, java.net.http], conf/management/jmxremote.password.template=[jdk.management.jfr, jdk.management, jdk.unsupported]}
Error: Process completed with exit code 1.
```


### Anything else?

There is a bug ticket on the java bug tracker with the same error message, but targets JDK 11:
https://bugs.openjdk.org/browse/JDK-8207015

--- a/.github/workflows/mac-dmg.yml
+++ b/.github/workflows/mac-dmg.yml
@@ -31,7 +31,7 @@ jobs:
           output-suffix: x64
           xcode-path: '/Applications/Xcode_13.2.1.app'
           fuse-lib: macFUSE
-          openjfx-url: https://download2.gluonhq.com/openjfx/20.0.1/openjfx-20.0.1_osx-aarch64_bin-jmods.zip
+          openjfx-url: https://download2.gluonhq.com/openjfx/20.0.1/openjfx-20.0.1_osx-x64_bin-jmods.zip
           openjfx-sha: 4fcd4bc3cd0edeb899108109e42a0c5a2d87d14a195d11199060862eb6d887b5
         - os: [self-hosted, macOS, ARM64]
           architecture: aarch64
@@ -54,7 +54,7 @@ jobs:
         id: download-jmods
         run: |
           curl -L ${{ matrix.openjfx-url }} -o openjfx-jmods.zip
-          echo "${{ matrix.openjfx-sha }} openjfx-jmods.zip" | sha256sum --check
+          echo "${{ matrix.openjfx-sha }} *openjfx-jmods.zip" | shasum -a256 --check
           mkdir -p openjfx-jmods/
           unzip -j openjfx-jmods.zip \*/javafx.base.jmod \*/javafx.controls.jmod \*/javafx.fxml.jmod \*/javafx.graphics.jmod -d openjfx-jmods
       - name: Ensure major jfx version in pom and in jmods is the same

--- a/.github/workflows/mac-dmg.yml
+++ b/.github/workflows/mac-dmg.yml
@@ -31,11 +31,15 @@ jobs:
           output-suffix: x64
           xcode-path: '/Applications/Xcode_13.2.1.app'
           fuse-lib: macFUSE
+          openjfx-url: https://download2.gluonhq.com/openjfx/20.0.1/openjfx-20.0.1_osx-aarch64_bin-jmods.zip
+          openjfx-sha: 4fcd4bc3cd0edeb899108109e42a0c5a2d87d14a195d11199060862eb6d887b5
         - os: [self-hosted, macOS, ARM64]
           architecture: aarch64
           output-suffix: arm64
           xcode-path: '/Applications/Xcode_13.2.1.app'
           fuse-lib: FUSE-T
+          openjfx-url: https://download2.gluonhq.com/openjfx/20.0.1/openjfx-20.0.1_osx-aarch64_bin-jmods.zip
+          openjfx-sha: e7e99e6dc3d091e7e1c6940d8e1acc282f22b82b234a20ae7cbec4b93a6acabe
     steps:
       - uses: actions/checkout@v3
       - name: Setup Java
@@ -49,11 +53,10 @@ jobs:
       - name: Download OpenJFX jmods
         id: download-jmods
         run: |
-          curl -L ${{ env.OPENJFX_JMODS_URL }} -o openjfx-jmods.zip
+          curl -L ${{ matrix.openjfx-url }} -o openjfx-jmods.zip
+          echo "${{ matrix.openjfx-sha }} openjfx-jmods.zip" | sha256sum --check
           mkdir -p openjfx-jmods/
           unzip -j openjfx-jmods.zip \*/javafx.base.jmod \*/javafx.controls.jmod \*/javafx.fxml.jmod \*/javafx.graphics.jmod -d openjfx-jmods
-        env:
-          OPENJFX_JMODS_URL: 'https://download2.gluonhq.com/openjfx/20.0.1/openjfx-20.0.1_osx-${{ matrix.architecture }}_bin-jmods.zip'
       - name: Ensure major jfx version in pom and in jmods is the same
         run: |
           JMOD_VERSION=$(jmod describe openjfx-jmods/javafx.base.jmod | head -1)

--- a/.github/workflows/mac-dmg.yml
+++ b/.github/workflows/mac-dmg.yml
@@ -80,7 +80,7 @@ jobs:
           ${JAVA_HOME}/bin/jlink
           --verbose
           --output runtime
-          --module-path "${JAVA_HOME}/jmods"
+          --module-path "${JAVA_HOME}/jmods:openjfx-jmods"
           --add-modules java.base,java.desktop,java.instrument,java.logging,java.naming,java.net.http,java.scripting,java.sql,java.xml,javafx.base,javafx.graphics,javafx.controls,javafx.fxml,jdk.unsupported,jdk.crypto.ec,jdk.accessibility,jdk.management.jfr
           --strip-native-commands
           --no-header-files

--- a/.github/workflows/mac-dmg.yml
+++ b/.github/workflows/mac-dmg.yml
@@ -41,21 +41,32 @@ jobs:
       - name: Setup Java
         uses: actions/setup-java@v3
         with:
-          distribution: 'zulu'
+          distribution: 'temurin'
           java-version: ${{ env.JAVA_VERSION }}
-          java-package: 'jdk+fx'
+          java-package: 'jdk'
           architecture: ${{ matrix.architecture }}
           cache: 'maven'
-      - name: Ensure major jfx version in pom equals in jdk
-        if: ${{ !contains(matrix.os, 'self-hosted') }}
-        shell: pwsh
+      - name: Download OpenJFX jmods
+        id: download-jmods
         run: |
-          $jfxPomVersion = (&mvn help:evaluate "-Dexpression=javafx.version" -q -DforceStdout) -split "\."
-          $jfxJdkVersion = ((Get-Content -path "${env:JAVA_HOME}/lib/javafx.properties" | Where-Object {$_ -like 'javafx.version=*' }) -replace '.*=','') -split "\."
-          if ($jfxPomVersion[0] -ne $jfxJdkVersion[0]) {
-            Write-Error "Major part of JavaFX version in pom($($jfxPomVersion[0])) does not match the version in JDK($($jfxJdkVersion[0])) "
+          curl -L ${{ env.OPENJFX_JMODS_URL }} -o openjfx-jmods.zip
+          mkdir -p openjfx-jmods/
+          unzip -j openjfx-jmods.zip \*/javafx.base.jmod \*/javafx.controls.jmod \*/javafx.fxml.jmod \*/javafx.graphics.jmod -d openjfx-jmods
+        env:
+          OPENJFX_JMODS_URL: 'https://download2.gluonhq.com/openjfx/20.0.1/openjfx-20.0.1_osx-${{ matrix.architecture }}_bin-jmods.zip'
+      - name: Ensure major jfx version in pom and in jmods is the same
+        run: |
+          JMOD_VERSION=$(jmod describe openjfx-jmods/javafx.base.jmod | head -1)
+          JMOD_VERSION=${JMOD_VERSION#*@}
+          JMOD_VERSION=${JMOD_VERSION%%.*}
+          POM_JFX_VERSION=$(mvn help:evaluate "-Dexpression=javafx.version" -q -DforceStdout)
+          POM_JFX_VERSION=${POM_JFX_VERSION#*@}
+          POM_JFX_VERSION=${POM_JFX_VERSION%%.*}
+
+          if [ $POM_JFX_VERSION -ne $JMOD_VERSION ]; then
+            >&2 echo "Major JavaFX version in pom.xml (${POM_JFX_VERSION}) != jmod version (${JMOD_VERSION_AMD64})"
             exit 1
-          }
+          fi
       - name: Set version
         run : mvn versions:set -DnewVersion=${{ needs.get-version.outputs.semVerStr }}
       - name: Run maven

